modules/imgproc/src/opencl/box_filter.cl

```diff
@@ -0,0 +1,78 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+__kernel void box_filter(__global const uint* src, int ssy, int srcOffsetX, int srcOffsetY,
+                         __global uint* dst, int dsy, int dst_offset, int rows, int cols
+#ifdef NORMALIZE
+                         , float alpha
+#endif
+                         )
+{
+    int x = get_global_id(0);
+    int y = get_global_id(1) * 2;
+    int ssx, dsx;
+
+    if ((x * 16) >= cols || y >= rows) return;
+
+    uint4 line[4];
+    uint4 line_out[2];
+    ushort a; ushort16 b; ushort c;
+    ushort d; ushort16 e; ushort f;
+    ushort g; ushort16 h; ushort i;
+    ushort j; ushort16 k; ushort l;
+
+    ssx = dsx = 1;
+    int src_index = x * 4 * ssx + (y - 1) * (ssy / 4);
```

Comment:
Could you add check that `(ssy % 4) == 0`? `isSubmatrix()` is check for borders only. But it doesn't guarantee that `cols * elemSize == step`.

Could you use these names (like other kernels):
`ssy -> src_step`
`dsy -> dst_step`